k8s/upgrade-job/src/helm/upgrade.rs

```diff
@@ -138,6 +138,10 @@ impl HelmUpgradeBuilder {
         let from_version = upgrade::path::version_from_release_chart(chart)?;
         let upgrade_path_is_valid =
             upgrade::path::is_valid(chart_variant.clone(), &from_version, &to_version)?;
+        ensure!(
+            to_version.ne(&from_version),
+            InstalledVersionSameAsUpgradeVersion
```

Comment:
If an upgrade is in-progress and user invokes command in another shell, will it show this error or status of running upgrade?

Reply:
In progess helm upgrades recede the release entry from the list of helm releases on that Kubernetes namespace. And before we upgrade, we validate the helm release. We'd not be able to find it on one of the two parallel upgrade runs (this is obviously not perfect, and depends on the reconciliation of Kubernetes' controllers). So, we should be safe.

Reply:
If the user, runs the parallel upgrade post-helm-upgrade, and during or pre-data-plane-restart, it will show this error.